Compile error with recent ffmpeg changes

I get the following with the newest compiler on macos:
```
../nginx-vod-module/vod/filters/audio_decoder.c:14:16: error: assigning to 'AVCodec *' (aka 'struct AVCodec *') from 'const AVCodec *' (aka 'const struct AVCodec *') discards qualifiers [-Werror,-Wincompatible-pointer-types-discards-qualifiers]
        decoder_codec = avcodec_find_decoder(AV_CODEC_ID_AAC);
                      ^ ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
```

It seems like a sensible error as the const returned from the function is assigned to a static that might break the const promise?

--- a/vod/filters/audio_decoder.c
+++ b/vod/filters/audio_decoder.c
@@ -1,7 +1,7 @@
 #include "audio_decoder.h"
 
 // globals
-static AVCodec *decoder_codec = NULL;
+static const AVCodec *decoder_codec = NULL;
 static bool_t initialized = FALSE;
 
 void

--- a/vod/filters/audio_encoder.c
+++ b/vod/filters/audio_encoder.c
@@ -14,11 +14,11 @@ typedef struct
 } audio_encoder_state_t;
 
 // globals
-static AVCodec *encoder_codec = NULL;
+static const AVCodec *encoder_codec = NULL;
 static bool_t initialized = FALSE;
 
 static bool_t
-audio_encoder_is_format_supported(AVCodec *codec, enum AVSampleFormat sample_fmt)
+audio_encoder_is_format_supported(const AVCodec *codec, enum AVSampleFormat sample_fmt)
 {
 	const enum AVSampleFormat *p;
 

--- a/vod/thumb/thumb_grabber.c
+++ b/vod/thumb/thumb_grabber.c
@@ -47,8 +47,8 @@ typedef struct {
 } codec_id_mapping_t;
 
 // globals
-static AVCodec *decoder_codec[VOD_CODEC_ID_COUNT];
-static AVCodec *encoder_codec = NULL;
+static const AVCodec *decoder_codec[VOD_CODEC_ID_COUNT];
+static const AVCodec *encoder_codec = NULL;
 
 static codec_id_mapping_t codec_mappings[] = {
 	{ VOD_CODEC_ID_AVC, AV_CODEC_ID_H264, "h264" },
@@ -60,7 +60,7 @@ static codec_id_mapping_t codec_mappings[] = {
 void
 thumb_grabber_process_init(vod_log_t* log)
 {
-	AVCodec *cur_decoder_codec;
+	const AVCodec *cur_decoder_codec;
 	codec_id_mapping_t* mapping_cur;
 	codec_id_mapping_t* mapping_end;
 

